src/components/annotations/draw.js

```diff
@@ -69,7 +69,7 @@ function drawOne(gd, index) {
     fullLayout._infolayer.selectAll('.annotation[data-index="' + index + '"]').remove();
 
     // remember a few things about what was already there,
-    var optionsIn = layout.annotations[index],
+    var optionsIn = (layout.annotations || [])[index],
```

Comment:
Ha. I'm looking forward to making annotations and shapes update _d3-idiomatic_. These user / full container double standard headaches will be gone!